transports/p2p/p2psync/manager.go

```diff
@@ -602,11 +603,16 @@ func (sm *SyncManager) handleHeadersMsg(hmsg *headersMsg) {
 		return
 	}
 
+	// If we received only blocks that we already have, or are not from the longest chain,
```

Comment:
I am checking it, but not directly. So, in this file, inside the for loop, in line 565, if the header is from the longest chain, the `finalHeight` variable is being set to its height. Otherwise it's set to 0. And 0 is also always lower than the current longest chain, so this if-statement also includes this check indirectly.

Maybe it should be refactored, bo for now I tried to do it the minimal way, to not mess up with the existing code too much.